src/app/core/services/authentication.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import {
+  BehaviorSubject,
+  catchError,
+  from,
+  map,
+  Observable,
+  tap,
+  throwError,
+} from 'rxjs';
+import firebase from 'firebase/compat';
+import FirebaseError = firebase.FirebaseError;
+import { UsersService } from './users.service';
+import { Store } from '@ngrx/store';
+import { removeToken, setToken } from '../states/auth/auth.actions';
+import { resolveError, setErrorMessage } from '../states/error/error.actions';
+import UserCredential = firebase.auth.UserCredential;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticationService {
+  loggedIn = new BehaviorSubject<boolean>(false);
+  loggedIn$ = this.loggedIn.asObservable();
```

Comment:
behavior subject + asObservable pattern is good 👍
FYI: You can replace this with a single loggedIn signal in the future